types/mongodb/index.d.ts

```diff
@@ -2433,9 +2433,9 @@ export interface GridFSBucketOptions {
     readPreference?: ReadPreferenceOrMode;
 }
 
-/** http://mongodb.github.io/node-mongodb-native/3.1/api/GridFSBucket.html#~errorCallback */
-export interface GridFSBucketErrorCallback {
-    (err?: MongoError): void;
+/** http://mongodb.github.io/node-mongodb-native/3.6/api/GridFSBucket.html#~errorCallback */
+export interface GridFSBucketErrorCallback<T = any> extends MongoCallback<T> {
+    (error: MongoError, result: T): void;
```

Comment:
Since you removed `?` I think that you should make this `| undefined`. The same goes for `T` since it won't be available when error is.

```suggestion
    (error: MongoError | undefined, result: T | undefined): void;
```

Maybe even, if it works:

```suggestion
    (error: undefined, result: T): void;
    (error: MongoError, result: undefined): void;
```

Reply:
Actually, since this is only used in one place (right?), I think it would be better to just:

```suggestion
    (error?: MongoError | null): void;
```

also here, I think that `null` is actually the value received, not undefined. I might be wrong though